sky/packages/sky/lib/sky_tool

```diff
@@ -275,25 +291,73 @@ class StopTracing(object):
 
 
 class SkyShellRunner(object):
+    def _update_paths(self):
+        global ADB_PATH
+        if 'ANDROID_HOME' in os.environ:
+            android_home_dir = os.environ['ANDROID_HOME']
+            ADB_PATH = os.path.join(android_home_dir, 'sdk/platform-tools/adb')
+
     def _check_for_adb(self):
         try:
-            subprocess.check_output([ADB_PATH, 'devices'])
+            adb_version = subprocess.check_output([ADB_PATH, 'version'])
+            # Sample output: "Android Debug Bridge version 1.0.31"
+            version_fields = adb_version.rstrip().split('.')
+            # If the string doesn't match the expected format, then skip the
+            # version check.
+            if len(version_fields) == 3 and version_fields[-1].isdigit():
+                minor_version = int(version_fields[-1])
+                if minor_version < 32:
```

Comment:
Please check the other two version numbers.  Otherwise we'll think that 1.1.2 is too old.